### ISSUE TITLE
Basic features and some ES6 doesn't work.

### ISSUE THREAD

**Issue by RobbieGM** [Khan/live-editor#651]
I would really like if we could use all of this in our Khan Academy code:
`
function f() { // Very basic. Yet Oh Noes prefers us writing it the messier way.
}
async function getButtonPressed() { // No, I didn't mean to type "asin", Oh Noes.
     return (await buttonClickedPromise).name;
}
class HowAboutThis { // It would also be nice if we could do elegant classes like this.
    constructor(x) {
        this.x = x;
    }
}
let thisDoesntWork = true;
const neitherDoesThis = true;

square = x => x * x; // Arrow functions would also be good

let enhancedObjectLiteral = {
    value: 3,
    square, // Easy shorthand for square: square,
    getDoubleValue() { return value * 2; } // Easy methods read like functions
};

function defaultParametersAreVeryUseful(a, isCorrect=true, b, c) {}
function restParametersAreBetterThanArgumentsList(a, b, ...rest) {}

for (let x of iterable) {
    // much better than "for (var i = 0; i < iterable.length; i++) { iterable[i] }
}

function * getPositiveIntegers(x=0) {
    yield x + 1;
}

// Thanks for your consideration. Teaching new programmers ES6 features will not only help their code be better, it will make sure they are up to date with modern JavaScript when they eventually get a job.
`

**Comment by JettBurns14** [comment]
> Currently we only support ES5. It would be nice to support ES6 at some point.

From #629. 

This _old_ Live Editor project has been receiving zero resources from the team for a good while, and probably won't be updated until they focus on the Computing section again. Just typical organization priorities and focus!

FYI, the [webpage environment](https://www.khanacademy.org/cs/new/webpage) offers a little more freedom. You can use function declarations, `let`, `const`, and a few other things I may have left out. 

I do agree with your request though, gave you a thumbs up. 😃

**Comment by Ben-Rivers** [comment]
If you really want to use some of the more basic features in the live editor (like me), you can put `jshint ignore:start` at the beginning of your program.

**Comment by JettBurns14** [comment]
Hello @txjoe, unfortunately this repository is no longer receiving attention from the KA engineers (see [README](https://github.com/Khan/live-editor)); and nope, it hasn't been moved. 

Since you still might want to contribute to the Live Editor anyway, I recommend you look through the [open issues](https://github.com/Khan/live-editor/issues) if you haven't already and see if you can fix some things locally, and create pull request(s) with your working changes. I'm pretty sure all setup instructions are in the README.

Hope this helps, I plan to begin submitting my own fixes in the Summer, perhaps we can work together later on!

**Comment by txjoe** [comment]
The readme section advices against submitting PRs though.

**Comment by JettBurns14** [comment]
This is true, but in my opinion, if you submit good code that works and thus doesn't take long to review, there's a chance it could be merged. No hurt in trying!

**Comment by naclcaleb** [comment]
I'm working on figuring out how to get JSHint to allow it.

The BabyHint parser was easy to hack; just changed some regexes and good to go. The global variables seem to be defined using JSHint.

 Please correct me if I'm wrong, because the jshint.js file is not pretty to look at!
